From: Patrick Brunschwig <patrick@enigmail.net>
Date: Tue, 13 Feb 2018 13:07:31 +0100
Subject: - sanitize email address in getSecretKeyByEmail() - determine key ID
 for From-address via getSecretKeyByEmail()

(cherry picked from commit c3597d663e6c3e2d5971fe08006b8eb1e77ebcc8)
---
 package/keyRing.jsm                     | 3 +++
 ui/content/enigmailMsgComposeOverlay.js | 7 +++++++
 2 files changed, 10 insertions(+)

diff --git a/package/keyRing.jsm b/package/keyRing.jsm
index 8af14aa..5c8893e 100644
--- a/package/keyRing.jsm
+++ b/package/keyRing.jsm
@@ -304,6 +304,9 @@ var EnigmailKeyRing = {
    * @return KeyObject with the found key, or null if no key found
    */
   getSecretKeyByEmail: function(emailAddr) {
+    // sanitize email address
+    emailAddr = emailAddr.replace(/([\.\[\]\-\\])/g, "\\$1");
+
     let searchTerm = "(<" + emailAddr + ">| " + emailAddr + "$|^" + emailAddr + "$)";
 
     return this.getSecretKeyByUserId(searchTerm);
diff --git a/ui/content/enigmailMsgComposeOverlay.js b/ui/content/enigmailMsgComposeOverlay.js
index c3d6ef1..0e2acb3 100644
--- a/ui/content/enigmailMsgComposeOverlay.js
+++ b/ui/content/enigmailMsgComposeOverlay.js
@@ -2569,6 +2569,13 @@ Enigmail.msg = {
       EnigmailLog.DEBUG("enigmailMsgComposeOverlay.js: Enigmail.msg.getSenderUserId: type of userIdValue=" + typeof(userIdValue) + "\n");
       userIdValue = this.identity.email;
     }
+
+    if (this.identity.getIntAttribute("pgpKeyMode") === 0) {
+      let key = EnigmailKeyRing.getSecretKeyByEmail(userIdValue);
+      if (key) {
+        userIdValue = "0x" + key.fpr;
+      }
+    }
     return userIdValue;
   },
 
